sunpy/map/tests/test_map_factory.py

```diff
@@ -50,6 +51,12 @@ def test_two_map_inputs(args1, args2):
         assert isinstance(out, sunpy.map.GenericMap)
 
 
+def test_read_asdf_and_verify(tmpdir):
+    loaded_asdf_map = sunpy.map.Map(AIA_ASDF)
+    assert isinstance(loaded_asdf_map.data, np.ndarray)
+    assert isinstance(loaded_asdf_map.meta, dict)
```

Comment:
You should also assert the type of the new object:
```suggestion
    assert isinstance(loaded_asdf_map.meta, dict)
    assert isinstance(loaded_asdf_map, sunpy.map.AIAMap)
```